### DISCUSSION TITLE
Send a unique ID for the dashboard to data source

### DISCUSSION THREAD

**Discussion by bboreham** [grafana/grafana#42541] [General]
While running a back-end data store for metrics, logs or traces, occasionally there will be a situation where lots of processes start crashing and failing all requests because someone sent in a particular request that doesn't sit well with the current implementation.

It's quite likely that this "poison request" comes from a particular dashboard (or Explore view*), and if that dashboard is on auto-refresh then it will keep coming and causing crashes until we can fix the back-end code, or perhaps block the IP address or the entire customer.

So, my request is to send an opaque UUID that doesn't reveal anything about what the customer is doing, but does let us identify and block that one dashboard until we can recover.

\* What about Explore view?  Since these are dynamically created, the above doesn't work.  Maybe the UUID could be a hash of the queries?

**Comment by dannykopping** [comment]
Perhaps we can keep it simple to start, and hash the URL of either the dashboard or the Explore view; that should give us enough specificity to block the requests.

If we move forward with this (and I hope we do because it's going to become more and more crucial), the data store should return an error which Grafana can identify and display a friendly error message (perhaps a `529 Site is overloaded` or something more standard like `503 Service Unavailable` with a message). We want the end-user experience to be quite clear with what's happening. Something along the lines of "Request blocked for troubleshooting purposes, please contact support" would probably express the intent clearly.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
